ggml-cuda.cu

```diff
@@ -6969,6 +6969,9 @@ void ggml_cuda_assign_scratch_offset(struct ggml_tensor * tensor, size_t offset)
     if (g_scratch_size == 0) {
         return;
     }
+
+    ggml_cuda_set_device(g_main_device);
+
```

Comment:
Could this be moved inside the `if` block below? It seems to be the only CUDA op relevant here.